src/metabase/models/params/chain_filter.clj

```diff
@@ -113,28 +125,21 @@
      (types/temporal-field? (t2/select-one [Field :base_type :semantic_type] :id field-id)))
    :ttl/threshold (u/minutes->ms 10)))
 
-(defn- filter-clause
+(mu/defn ^:private filter-clause
   "Generate a single MBQL `:filter` clause for a Field and `value` (or multiple values, if `value` is a collection)."
-  [source-table-id field-id value]
+  [source-table-id
+   {:keys [field-id op value options]} :- Constraint]
   (let [field-clause (let [this-field-table-id (field/field-id->table-id field-id)]
                        [:field field-id (when-not (= this-field-table-id source-table-id)
                                           {:join-alias (joined-table-alias this-field-table-id)})])]
-    (cond
-      ;; e.g. {$$venues.price [:between 2 3]} -> [:between $venues.price 2 3]
-      ;; this is not really supported by the API directly
-      (and (sequential? value) (keyword? (first value)))
-      (into [(first value) field-clause] (rest value))
-      ;; e.g. {$$venues.price #{2 3}} -> [:= $$venues.price 2 3]
-      (and (coll? value) (not (map? value)))
-      (into [:= field-clause] value)
-      :else
-      ;; e.g. {$$venues.price "past32weeks"} -> [:time-interval $checkins.date -32 :week]
-      (or (when (and (temporal-field? field-id)
-                     (string? value))
-            (u/ignore-exceptions
-              (params.dates/date-string->filter value field-id)))
-          ;; e.g. {$$venues.price 2} -> [:= $$venues.price 2]
-          [:= field-clause value]))))
+    (if (and (temporal-field? field-id)
+             (string? value))
+      (u/ignore-exceptions
+        (params.dates/date-string->filter value field-id))
+      (cond-> [op field-clause]
+        ;; we don't want to skip our value, even if its nil
```

Comment:
can you clarify wdym about this? do you mean when the value is set as `nil`?

if so do we have a test?

Reply:
`(u/one-or-many nil)` returns `nil`, and `(into [op field] nil)` is `[op field]`, while we need `[op field nil]` (with the value).

And yes, tests fail a lot on that one. :)